images/macos/provision/configuration/CreateCleanAnkaTemplate.ps1

```diff
@@ -0,0 +1,232 @@
+        Start-Sleep -Seconds $RetryInterval
+    }
+
+    Write-Host "All the attempts exhausted, try again later"
+    exit 1
+}
+
+function New-AnkaVMTemplate {
+    param (
+        [String] $InstallerPath,
+        [String] $ShortMacOSVersion,
+        [String] $TemplateName,
+        [String] $TemplateUsername,
+        [String] $TemplatePassword,
+        [Int] $CpuCount,
+        [Int] $RamSizeGb,
+        [Int] $DiskSizeGb,
+        [String] $DisplayResolution
+    )
+
+    $isTemplateExists = (Invoke-Anka { anka --machine-readable list } | ConvertFrom-Json).body.name -eq $templateName
+    if ($isTemplateExists)
+    {
+        Write-Host "Deleting existed template with name '$templateName' before creating a new one"
+        Invoke-Anka { anka delete $templateName --yes }
+    }
+
+    Write-Host "Creating Anka VM template with name '$templateName' and user $TemplateUsername"
+    $env:ANKA_DEFAULT_USER = $TemplateUsername
+    $env:ANKA_DEFAULT_PASSWD = $TemplatePassword
+    $env:ANKA_CREATE_SUSPEND = 0
+    Invoke-Anka { anka create --cpu-count $CpuCount --ram-size "${RamSizeGb}G" --disk-size "${DiskSizeGb}G" --app $InstallerPath $templateName }
+
+    # Apple Metal is available starting from Big Sur
+    if (-not $ShortMacOSVersion.StartsWith("10"))
+    {
+        Write-Host "Enabling Graphics Acceleration with Apple Metal for $templateName"
+        Invoke-Anka { anka modify $templateName set display -c pg }
+    }
+    Write-Host "Setting screen resolution to $DisplayResolution for $templateName"
+    Invoke-Anka { anka modify $templateName set display -r $DisplayResolution }
+
+    return $templateName
+}
+
+function Add-AnkaImageToRegistry {
+    param (
+        [String] $RegistryUrl,
+        [String] $ShortMacOSVersion,
+        [String] $TemplateName
+    )
+
+    Write-Host "Adding Anka remote registry to the list of registries"
+    Invoke-Anka { anka registry add ankaregistry $RegistryUrl  }
+
+    Write-Host "Setting up default registry for push"
+    Invoke-Anka { anka registry set ankaregistry }
+
+    Write-Host "Pushing clean macOS $ShortMacOSVersion to the $RegistryUrl"
+    Invoke-Anka { anka registry push -t $ShortMacOSVersion $TemplateName }
+}
+
+function Invoke-Anka {
+    param (
+        [scriptblock] $Cmd
+    )
+
+    & $Cmd
+    if ($LASTEXITCODE -ne 0) {
+        Write-Error "There is an error during command execution"
+        exit 1
+    }
+}
+
+function Get-ShortMacOSVersion {
+    param (
+        [Version] $MacOSVersion
+    )
+
+    # Take Major.Minor version for macOS 10 (10.14 or 10.15) and Major for all further versions
+    if ($MacOSVersion.Major -eq 10)
+    {
+        $shortMacOSVersion = "$($MacOSVersion.Major).$($MacOSVersion.Minor)"
+    }
+    else
+    {
+        $shortMacOSVersion = "$($MacOSVersion.Major)"
+    }
+
+    return $shortMacOSVersion
+}
+
+$macOSInstaller = (Get-MacOSInstaller -MacOSVersion $MacOSVersion -BetaSearch $BetaSearch)[-1]
```

Comment:
May be move `[-1]` logic to the `Get-MacOSInstaller` level via `Select-Object -Last 1`?
Or rename `Get-MacOSInstaller` -> `Get-MacOSInstallers`